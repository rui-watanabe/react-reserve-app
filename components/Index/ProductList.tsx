import { Card } from 'semantic-ui-react';
import { ProductListProps } from './IndexType';
import { ProductModelType } from '../../models/ProductModel/ProductType';

function ProductList({ products }: ProductListProps): JSX.Element {
  function mapProductsToItems(products: ProductModelType[]) {
    return products.map((product) => ({
      header: product.name,
      image: product.mediaUrl,
      meta: `$${product.price}`,
      color: 'teal',
      fluid: true,
      childKey: product._id,
      href: `/product?_id=${product._id}`,
    }));
  }

  return (
    <Card.Group
      stackable
      itemsPerRow="3"
      centered
      items={mapProductsToItems(products)}
    />
  );
}

export default ProductList;
