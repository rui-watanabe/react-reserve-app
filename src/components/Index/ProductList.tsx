import React from 'react';
import { Card } from 'semantic-ui-react';
import { ProductListProps } from '../Types/IndexType';
import { ProductModelType } from '../../../models/Types/ProductType';

function ProductList({ products }: ProductListProps): JSX.Element {
  function mapProductsToItems(products: ProductModelType[]) {
    return products.map((product) => ({
      header: product.name,
      image: product.mediaUrl,
      meta: `$${product.price}`,
      color: 'teal',
      fluid: true,
      childKey: product._id,
      href: `/product?_id=${product._id}`,
    }));
  }

  return (
    <Card.Group
      stackable
      itemsPerRow="3"
      centered
      items={mapProductsToItems(products)}
    />
  );
}

export default ProductList;
