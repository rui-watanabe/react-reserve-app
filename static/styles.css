:root {
  --bodyBg: #f0f4f8;
  --menuBg: #008080;
  --progressBar: #54d1db;
}

body {
  background: var(--bodyBg);
}

#menu {
  background: var(--menuBg);
}

/* Overrides default styled accordion styles (too much bottom padding) */
.ui.styled.accordion .content {
  margin: 0;
  padding: 0.5em 1em 0.75em !important;
}

.ui.card > .image:not(.ui) > img, .ui.cards > .card > .image:not(.ui) > img  {
  height: 212px;
}
