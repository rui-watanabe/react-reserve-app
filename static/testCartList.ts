import { CartProductType } from '../src/pagesTypes/CartType';
import { testProductList } from './testProductList';

export const testCartList: CartProductType = [
  {
    quantity: 1,
    product: testProductList[0],
  },
  {
    quantity: 2,
    product: testProductList[1],
  },
];
