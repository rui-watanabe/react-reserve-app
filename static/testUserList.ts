import { ObjectId } from 'mongodb';
import { UserModelType } from '../models/Types/UserType';

export const testUserList: UserModelType[] = [
  {
    _id: new ObjectId('000000000000000000000001'),
    createdAt: new Date('2020-9-22'),
    updatedAt: new Date('2020-9-22'),
    name: 'test',
    email: 'test@gmail.com',
    password: '11111111',
    role: 'root',
  },
  {
    _id: new ObjectId('000000000000000000000002'),
    createdAt: new Date('2020-9-22'),
    updatedAt: new Date('2020-9-22'),
    name: 'test2',
    email: 'test2@gmail.com',
    password: '22222222',
    role: 'admin',
  },
  {
    _id: new ObjectId('000000000000000000000003'),
    createdAt: new Date('2020-9-22'),
    updatedAt: new Date('2020-9-22'),
    name: 'test',
    email: 'test@gmail.com',
    password: '33333333',
    role: 'user',
  },
];
