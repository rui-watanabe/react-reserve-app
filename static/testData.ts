import { ProductModelType } from '../models/Types/ProductType';
import { UserModelType } from '../models/Types/UserType';
import { CartProductType } from '../src/pagesTypes/CartType';
import { testProductList } from './testProductList';
import { testUserList } from './testUserList';

export const testProduct: ProductModelType = testProductList[0];

export const testTotalPages = Math.floor(testProductList.length / 9);

export const testUser: UserModelType = testUserList[2]; // 'user' role user

export const testCartProductList: CartProductType = [
  { product: { ...testProduct }, quantity: 1 },
  { product: { ...testProductList[1] }, quantity: 2 },
  { product: { ...testProductList[2] }, quantity: 3 },
];
