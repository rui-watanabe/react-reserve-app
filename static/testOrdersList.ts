import { ObjectId, ObjectID } from 'mongodb';
import { OrderModelType } from '../models/Types/OrderType';
import { testProductList } from './testProductList';
import { testUserList } from './testUserList';

export const testOrdersList: OrderModelType[] = [
  {
    _id: new ObjectId('200000000000000000000001'),
    createdAt: new Date(2020, 4, 1, 12, 30, 45),
    updatedAt: new Date(2020, 4, 1, 12, 30, 45),
    products: [
      {
        _id: new ObjectID('100000000000000000000001'),
        quantity: 1,
        product: testProductList[0],
      },
      {
        _id: new ObjectID('100000000000000000000002'),
        quantity: 2,
        product: testProductList[1],
      },
    ],
    user: testUserList[2]._id, // 'user' role user
    email: testUserList[2].email,
    total: testProductList[0].price + testProductList[1].price * 2,
  },
  {
    _id: new ObjectID('200000000000000000000002'),
    createdAt: new Date(2020, 4, 1, 12, 30, 45),
    updatedAt: new Date(2020, 4, 1, 12, 30, 45),
    products: [
      {
        _id: new ObjectID('100000000000000000000003'),
        quantity: 1,
        product: testProductList[2],
      },
      {
        _id: new ObjectID('100000000000000000000004'),
        quantity: 2,
        product: testProductList[3],
      },
    ],
    user: testUserList[2]._id, // 'user' role user
    email: testUserList[2].email,
    total: testProductList[2].price + testProductList[3].price * 2,
  },
  {
    _id: new ObjectID('200000000000000000000003'),
    createdAt: new Date(2020, 4, 1, 12, 30, 45),
    updatedAt: new Date(2020, 4, 1, 12, 30, 45),
    products: [
      {
        _id: new ObjectID('100000000000000000000005'),
        quantity: 1,
        product: testProductList[4],
      },
      {
        _id: new ObjectID('100000000000000000000006'),
        quantity: 2,
        product: testProductList[5],
      },
    ],
    user: testUserList[2]._id, // 'user' role user
    email: testUserList[2].email,
    total: testProductList[4].price + testProductList[5].price * 2,
  },
];
