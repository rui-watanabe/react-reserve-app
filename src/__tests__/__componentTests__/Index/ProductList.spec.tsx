import React from 'react';
import renderer from 'react-test-renderer';
import { testProductList } from '../../../../static/testProductList';
import ProductList from '../../../components/Index/ProductList';

test('ProductList', () => {
  const component = renderer.create(<ProductList products={testProductList} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
