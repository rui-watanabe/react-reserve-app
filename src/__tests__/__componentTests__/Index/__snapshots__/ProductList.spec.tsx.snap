// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList 1`] = `
<div
  className="ui centered stackable three cards"
>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000001"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-5.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/f/d/fd1_lngchr_bh_frontlow-field-lounge-chair-tait-blush.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Bernie Gledhill
      </div>
      <div
        className="meta"
      >
        $1399.02
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000002"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-3.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/d/s/ds1_lngchr_ca-deep-thoughts-lounge-chair-camel-leather.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Gwendolen Bickerstaffe
      </div>
      <div
        className="meta"
      >
        $1599.93
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000003"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-1.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/r/u/ru1_rton69_m1_overhead-right-on-6-9-rug-color-mix-1.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Desi Avramovitz
      </div>
      <div
        className="meta"
      >
        $699.57
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000004"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-3.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/c/l/cl1_secktc_cl_view1-cleon-small-sectional-tait-charcoal.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Addy Alldre
      </div>
      <div
        className="meta"
      >
        $1999.23
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000005"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-4.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/o/n/on1_sleepr_rd_one-night-stand-craig-red.2x.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Cristian Gilbanks
      </div>
      <div
        className="meta"
      >
        $1999.38
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000006"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-2.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/c/d/cd1-2dcrdz-wl_clad-2-door-credenza.2x.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Kalila Gooms
      </div>
      <div
        className="meta"
      >
        $1299.18
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000007"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-4.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/s/r/sr1_sidtbl_sk_front_strut-side-table-smoke_3.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Bartolemo Peckitt
      </div>
      <div
        className="meta"
      >
        $449.39
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000008"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-3.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/r/o/rook_modern_table_lamp.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Kathye Haith
      </div>
      <div
        className="meta"
      >
        $199.93
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000009"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-3.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/n/y/ny1_narmsf_ln_frontlow-9-yard-armless-sofa-sunbrella-linen.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Armin Basilio
      </div>
      <div
        className="meta"
      >
        $2299.01
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000010"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-4.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/b/p/bp1_smlpen_ox-bobber-small-pendant-oxblood.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Cal Sterman
      </div>
      <div
        className="meta"
      >
        $299.23
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000011"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-5.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/w/r/wr1_twwood_wl_front-woodrow-twin-walnut_1.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Ardine Imlaw
      </div>
      <div
        className="meta"
      >
        $1299.53
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000012"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-5.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/m/o/modu-licious_4_-_modern_dresser_storage_4.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Noella Ruddom
      </div>
      <div
        className="meta"
      >
        $999.21
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000013"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-5.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/d/k/dk1_lmbrpl_og_front.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Kirstyn Espadate
      </div>
      <div
        className="meta"
      >
        $99.57
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000014"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-5.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/n/s/ns1_lnchbk_dk_new-standard-lounge-chair-nixon-dusk-black.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Rourke Greet
      </div>
      <div
        className="meta"
      >
        $749.29
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000015"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-4.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/h/o/hotel-modern-rug-navy-black_1.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Scarlet Rosemallow
      </div>
      <div
        className="meta"
      >
        $399.57
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000016"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-2.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/s/n/sn1_smotto_bl_frontlow-sunday-small-ottoman-agnew-blue.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Arden Stockbridge
      </div>
      <div
        className="meta"
      >
        $349.08
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000017"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-2.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/b/t/bt1_dinchr_in-beween-us-dining-chair-ink-leather.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Sheffy Gunter
      </div>
      <div
        className="meta"
      >
        $499.06
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000018"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-5.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/j/v/jv1_coatrk_tm_frontlow-splash-coat-rack-tomato-white-oak.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Alexine Mulligan
      </div>
      <div
        className="meta"
      >
        $299.27
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000019"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-1.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/d/k/dk1_sqrplw_mg_front.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Micheline Charlson
      </div>
      <div
        className="meta"
      >
        $119.18
      </div>
    </div>
  </a>
  <a
    className="ui teal fluid card"
    href="/product?_id=100000000000000000000020"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="https://res-3.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/c/i/ci1_lgcoff_gg_frontlow_v2-circula-large-coffee-table-grey-green.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Hughie Jeffryes
      </div>
      <div
        className="meta"
      >
        $1199.57
      </div>
    </div>
  </a>
</div>
`;
